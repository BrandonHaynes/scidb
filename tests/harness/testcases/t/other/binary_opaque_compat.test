--setup 
--start-query-logging
--start-ignore-warnings

create array SortedPrevious <d:int64> [ X=0:*,10000,0]

--test
# load the array from the opaque file saved by the previous version of SciDB
--shell --store-all --command "iquery -c $IQUERY_HOST -p $IQUERY_PORT -naq "load(SortedPrevious, '${TEST_DATA_DIR}/previous.out', -2, 'opaque')""

# check that the size is correct and that the data is sorted
aggregate(SortedPrevious, count(d))
--igdata "store(subarray(SortedPrevious, 1000, 1100), Slice)"
--igdata "store(window(Slice, 0, 1, min(d)), Window)"
aggregate(filter(join(Slice, Window), d!=d_min), count(*))

--cleanup

remove(SortedPrevious)
remove(Slice)
remove(Window)

--stop-ignore-warnings
--stop-query-logging
