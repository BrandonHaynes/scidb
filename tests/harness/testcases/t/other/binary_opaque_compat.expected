SCIDB QUERY : <create array SortedPrevious <d:int64> [ X=0:*,10000,0]>
Query was executed successfully

SCIDB QUERY : <iquery -c $IQUERY_HOST -p $IQUERY_PORT -naq "load(SortedPrevious, '${TEST_DATA_DIR}/previous.out', -2, 'opaque')">
Query was executed successfully

SCIDB QUERY : <aggregate(SortedPrevious, count(d))>
{i} d_count
{0} 40001

SCIDB QUERY : <store(subarray(SortedPrevious, 1000, 1100), Slice)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(window(Slice, 0, 1, min(d)), Window)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <aggregate(filter(join(Slice, Window), d!=d_min), count(*))>
{i} count
{0} 0

SCIDB QUERY : <remove(SortedPrevious)>
Query was executed successfully

SCIDB QUERY : <remove(Slice)>
Query was executed successfully

SCIDB QUERY : <remove(Window)>
Query was executed successfully

